src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -132,11 +132,9 @@ internal ImmutableArray<FunctionMetadata> LoadFunctionMetadata(bool forceRefresh
             ImmutableArray<FunctionMetadata> immutableFunctionMetadata;
             var workerConfigs = _languageWorkerOptions.Value.WorkerConfigs;
 
-            IFunctionMetadataProvider metadataProvider = Utility.CanWorkerIndex(workerConfigs, _environment)
-                ? new WorkerFunctionMetadataProvider(_loggerFactory.CreateLogger<WorkerFunctionMetadataProvider>(), dispatcher)
-                : _functionMetadataProvider;
+            IFunctionMetadataProvider metadataProvider = new AggregateFunctionMetadataProvider(_loggerFactory.CreateLogger<AggregateFunctionMetadataProvider>(), dispatcher, _functionMetadataProvider);
 
-            immutableFunctionMetadata = metadataProvider.GetFunctionMetadataAsync(workerConfigs, forceRefresh).GetAwaiter().GetResult();
+            immutableFunctionMetadata = metadataProvider.GetFunctionMetadataAsync(workerConfigs, SystemEnvironment.Instance, forceRefresh).GetAwaiter().GetResult();
```

Comment:
FWIW this is another path that should be async (and another in here not present in the PR diff) - can we change the interface to remedy these? If we're loading _all_ of these asynchronously as an end game (100% out of proc), we should be async on the paths.

Reply:
I agree with the issue on async pattern and it not being highlighted in PR diff. I will discuss with folks in the team about things to do to improve the pattern here.